File: 0783-minimum-distance-between-bst-nodes/0783-minimum-distance-between-bst-nodes.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minDiffInBST(self, root: Optional[TreeNode]) -> int:
        
        st = [root]
        arr = [] 
        while st: 
            node = st.pop()
            arr.append(node.val)
            if node.left: 
                st.append(node.left)
            if node.right:
                st.append(node.right)
        arr.sort() 
        ans = float('inf')
        for i in range(1,len(arr)): 
            if arr[i] - arr[i-1] < ans: 
                ans = arr[i] - arr[i-1]
        return ans 
            
            
            
    
    

File: 0783-minimum-distance-between-bst-nodes/0783-minimum-distance-between-bst-nodes.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> arr ; 
    
    void inorder(TreeNode* node){
        if(node ==NULL) return ; 
        
        
        inorder(node->left) ; 
        arr.push_back(node->val) ; 
        inorder(node->right) ;
    }
    
    
    
    int minDiffInBST(TreeNode* root) {
        
        inorder(root) ;
        int ans = INT_MAX ; 
        for(int i=1 ; i < arr.size() ; i++){
            if( arr[i]-arr[i-1] < ans ) 
                ans = arr[i] -arr[i-1] ; 
        }
        return ans ; 
    }
};